### ISSUE TITLE
[Feature] Add a cron schedule to an sqljob

### ISSUE THREAD

**Issue by chris102994** [mariadb-operator/mariadb-operator#110]
**Is your feature request related to a problem? Please describe.**
The problem I face is the need for a SQL cron job. An application I am using grows the database my logging to a table unfortunately. I would like to control this by adding a cron scheduled sqljob that can run daily.

**Describe the solution you'd like**
Here is my sqljob (Edited to remove info)

```yaml
---
apiVersion: mariadb.mmontes.io/v1alpha1
kind: SqlJob
metadata:
  name: delete-logs
  namespace: my-namespace
spec:
  mariaDbRef:
    name: mariadb
  username: mariadb
  database: mariadb
  passwordSecretKeyRef:
    name: mariadb-secret
    key: db_password
  sql: |
    DELETE FROM logs
    WHERE date < DATE_SUB(NOW(), INTERVAL 60 DAY);

```

I'd like to add `spec.schedule` to the CRD so that it can behave like a scheduled `Backup` rather than being a 1-shot job. i.e.

```yaml
---
apiVersion: mariadb.mmontes.io/v1alpha1
kind: SqlJob
metadata:
  name: delete-logs
  namespace: my-namespace
spec:
  mariaDbRef:
    name: mariadb
  schedule:
    cron: "0 3 * * *"
    suspend: false
  username: mariadb
  database: mariadb
  passwordSecretKeyRef:
    name: mariadb-secret
    key: db_password
  sql: |
    DELETE FROM logs
    WHERE date < DATE_SUB(NOW(), INTERVAL 60 DAY);

```

**Describe alternatives you've considered**
I considered building a pod that can run on a cron-schedule and apply/delete the above yaml after a success.

**Additional context**
* Cluster is a 3-node rancher k3s cluster.
* MariaDB is HA with replication.

**Environment details**:
- Kubernetes version: v1.21.5+k3s2
- mariadb-operator version: v0.0.12
- Install method: helm over argocd (chart version 0.12.0)
- Install flavour: recommended

**Comment by mmontes11** [comment]
Hey @chris102994 ! Thanks for your suggestion.

This shouldn't be too hard to implement, I would agree that periodically running a `SqlJob` can have a lot of use cases like yours.

In addition to your CRD definition, I would suggest adding some fields from the Job and CronJob spec to allow the user fully customize the underlying the Job like we do already with the `Backup`:
- https://github.com/mariadb-operator/mariadb-operator/blob/main/api/v1alpha1/backup_types.go#L55

```yaml
apiVersion: mariadb.mmontes.io/v1alpha1
kind: SqlJob
metadata:
  name: delete-logs
  namespace: my-namespace
spec:
  mariaDbRef:
    name: mariadb
  schedule:
    cron: "0 3 * * *"
    suspend: false
  backoffLimit: 5
  restartPolicy: OnFailure
  concurrencyPolicy: Forbit
  .... 
  sql: |
    DELETE FROM logs
    WHERE date < DATE_SUB(NOW(), INTERVAL 60 DAY);
```

It has been prioritized already in our roadmap. Contributions are very welcome, so feel free to give it a try if you like! Happy to help you if you need! 🙏🏻

**Comment by mmontes11** [comment]
@chris102994 @CanisHelix this has been supported by this [PR](https://github.com/mariadb-operator/mariadb-operator/pull/116) and will be released in the `v0.0.14` the next few days, stay tuned 🚀

**Comment by chris102994** [comment]
Awesome! Thanks!
